ocsgw/src/main/java/org/ostelco/ocsgw/data/grpc/GrpcDataSource.java

```diff
@@ -77,9 +81,22 @@ protected boolean removeEldestEntry(Map.Entry<String, SessionContext> eldest) {
     };
 
 
-    private abstract class AbstactObserver<T> implements StreamObserver<T> {
+    private abstract class CreditControlRequestObserver<T> implements StreamObserver<T> {
+        public final void onError(Throwable t) {
+            LOG.error("We got an error", t);
+            reconnectCreditControlRequest();
+        }
+
+        public final void onCompleted() {
+            // Nothing to do here
+            LOG.info("It seems to be completed");
```

Comment:
Looks unnecessary

Reply:
Yes I just wanted to check if it ever got called. So far it has never been. I will remove it